core/src/main/java/com/scalar/db/storage/rpc/GrpcScanOnBidirectionalStream.java

```diff
@@ -141,15 +141,15 @@ private void throwIfErrorForNext(ResponseOrError responseOrError) throws Executi
     }
   }
 
-  public void closeScanner() throws IOException {
+  public void closeScanner() throws ExecutionException {
     try {
       if (!hasMoreResults.get()) {
         return;
       }
-      requestObserver.onCompleted();
       hasMoreResults.set(false);
+      requestObserver.onCompleted();
```

Comment:
Just a question.
When users call `closeScanner` explicitly, this `onCompleted` will be called twice?
Is it OK to call `onCompleted` multiple times? (If OK, then I think there is no problem)